File: Pyhton/BioskopLuxury.py
from BioskopPremium import BioskopPremium

#Deklarasi class BioskopLuxury yang inheritage dengan class BioskopPremium
#class BioskopPremium inheritage dengan class Bioskop jadi bis dibilang ini multilevel inheritage
class BioskopLuxury(BioskopPremium) :
    #sebuah konstruktor untuk kelas ini dengan parameter berikut
    def __init__(self, id, nama, alamat, jumlah_studio, kota, nama_lounge, kapasitas_lounge, harga_tiket_premium, nama_restoran, jumlah_meja, jumlah_reservasi):
        super().__init__(id, nama, alamat, jumlah_studio, kota, nama_lounge, kapasitas_lounge, harga_tiket_premium)
        self.__nama_restoran = nama_restoran
        self.__jumlah_meja = jumlah_meja
        self.__jumlah_reservasi = jumlah_reservasi

    #berbagai method getset untuk setiap atribut
    def getnamarestoran(self):
        return self.__nama_restoran

    def setnamarestoran(self, nama_restoran):
        self.__nama_restoran = nama_restoran

    def getjumlahmeja(self):
        return self.__jumlah_meja

    def setjumlahmeja(self, jumlah_meja):
        self.__jumlah_meja = jumlah_meja

    def getjumlahreservasi(self):
        return self.__jumlah_reservasi

    def setjumlahreservasi(self, jumlah_reservasi):
        self.__jumlah_reservasi = jumlah_reservasi

    

File: C++/BioskopLuxury.cpp
#include <iostream>
#include <string>
#include "BioskopPremium.cpp"

using namespace std;

class BioskopLuxury : public BioskopPremium {
    private : 
        string nama_restoran;
        int jumlah_meja;
        int jumlah_reservasi;
    
    public :
        BioskopLuxury(){

        }

        BioskopLuxury(int id, string nama, string alamat, int jumlah_studio, string kota, 
        string nama_lounge, int kapasitas_lounge, double harga_tiket, 
        string nama_restoran, int jumlah_meja, int jumlah_reservasi) : BioskopPremium(id, nama, alamat, jumlah_studio, kota, nama_lounge, kapasitas_lounge, harga_tiket){
            this->nama_restoran = nama_restoran;
            this->jumlah_meja = jumlah_meja;
            this->jumlah_reservasi = jumlah_reservasi;
        }

        void setnamarestoran(string nama_restoran){
            this->nama_restoran = nama_restoran;
        }

        string getnamarestoran(){
            return this->nama_restoran;
        }

        void setjumlahmeja(int jumlah_meja){
            this->jumlah_meja = jumlah_meja;
        }

        int getjumlahmeja(){
            return this->jumlah_meja;
        }

        void setjumlahreservasi(int jumlah_reservasi){
            this->jumlah_reservasi = jumlah_reservasi;
        }

        int getjumlahreservasi(){
            return this->jumlah_reservasi;
        }

        
};